pkg/grpc/grpc_util.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2018 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package grpc
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/pkg/client/generated/cstor-volume-grpc/v1alpha1"
+	"github.com/openebs/maya/pkg/grpc/api"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+)
+
+// CmdSnaphotOptions holds the options for snapshot
+// create command
+type CmdSnaphotOptions struct {
```

Comment:
Can this be renamed to `snapshotOptions`? I felt it should be private struct. The instantiation can be `func SnapshotOptions() *snapshotOptions {}`

Reply:
Can this `snapshotOptions` embed the `CASOptions` that we may create at `pkg/apis/openebs.io/v1alpha1/`

Reply:
Let us have a discussion on this tomorrow.